numpy/core/setup.py

```diff
@@ -462,6 +476,10 @@ def generate_config_h(ext, build_dir):
 
             if can_link_svml():
                 moredefs.append(('NPY_CAN_LINK_SVML', 1))
+            
+            if can_link_highway():
+                moredefs.append(('NPY_CAN_LINK_HIGHWAY', 1))
+                moredefs.append(('VQSORT_ONLY_STATIC', 1))
```

Comment:
Is it necessary to define `HWY_COMPILE_ONLY_STATIC`, given that NumPy is in control of dynamic dispatch here? I am looking at https://github.com/google/highway/blob/master/g3doc/quick_reference.md#advanced-configuration-macros, and it seems to me that without doing that, we get the default which may include cpu-dependent targets (disclaimer: I haven't tried to verify this).

Reply:
Despite the similar naming, VQSORT_ONLY_STATIC is aimed at avoiding a dependency on .so/DLL, whereas HWY_COMPILE_ONLY_STATIC is only relevant when using Highway's multiple-target-compilation feature, which we are not here. FYI multiple-target-compilation requires opt-in from user code by including foreach_target.h.

Thus we do not have to define HWY_COMPILE_ONLY_STATIC.

Reply:
Thanks, it's clear to me now - resolving.